### PR TITLE
Fix: Prepare table component for assigment #6526

### LINKED ISSUE #6526: Prepare table component for assigment
We need table component based on cx-table with additional checkbox column, allows for assign items.
More here: https://cxwiki.sap.com/pages/viewpage.action?spaceKey=spar&title=Common+B2B+Commerce+Org+UX+-+Listings%2C+Assignments%2C+Viewing+and+Creating#CommonB2BCommerceOrgUX-Listings,Assignments,ViewingandCreating-Assigning(akaManageorAdd/RemoveorAssociations)

My purpose is to use 2 data sets (observables) as input:
- all items
- items assigned to managed subject 

Checkbox changes should emit event for parent component.

### PR DESCRIPTION
**Pull request by architworld**
Closes #6526

### CODE REVIEW

File: projects/storefrontlib/src/shared/components/checktable/checktable.component.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  @Input()
+  tableData: Array<any>;
+
+  @Output() toggle = new EventEmitter<any[]>();
+
+  selectedValues: Array<any>;
+
+  onToggle(event, row) {
+    // console.log(event.target.value)
+    if (event.target.value = 'on') {
```
Comment (Pucek9) [inline projects/storefrontlib/src/shared/components/checktable/checktable.component.ts]: why do you set that 'on' string to target.value? I think we need emit value on change value always after change `this.toggle.emit(event.target.value, row);`

File: projects/storefrontlib/src/shared/components/checktable/checktable.component.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  @Input()
+  tableData: Array<any>;
+
+  @Output() toggle = new EventEmitter<any[]>();
+
+  selectedValues: Array<any>;
+
+  onToggle(event, row) {
+    // console.log(event.target.value)
+    if ((event.target.value = 'on')) {
```
Comment (Pucek9) [inline projects/storefrontlib/src/shared/components/checktable/checktable.component.ts]: why do you set that 'on' string to target.value? I think we need emit value on change value always after change `this.toggle.emit(event.target.value, row);`

File: projects/storefrontlib/src/shared/components/checktable/checktable.component.html [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      </th>
+    </thead>
+    <tbody>
+      <tr *ngFor="let row of tableData">
+        <td class="cx-code" *ngFor="let column of columns">
+          <!-- <div class="d-md-none cx-label" *ngIf="column.checked===false">
+            {{ column.value }}
+          </div> -->
+
+          <div *ngIf="column.checked">
```
Comment (Pucek9) [inline projects/storefrontlib/src/shared/components/checktable/checktable.component.html]: I think we can add hardcoded **td** before <td class="cx-code" *ngFor="let column of columns"> without any text, and also add Assign **th** before th listing

File: projects/storefrontlib/src/shared/components/checktable/checktable.component.ts (hunk 2 above)
Comment (architworld) [reply projects/storefrontlib/src/shared/components/checktable/checktable.component.ts]: set as per the suggestion. Emit function can only take one argument, so i put it inside {}

File: projects/storefrontlib/src/shared/components/checktable/checktable.component.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  @Input()
+  tableData: Array<any>;
+
+  @Output() toggle = new EventEmitter<any>();
+
+  selectedValues: Array<any>;
+
+  onToggle(event, row) {
+    this.toggle.emit({ value: event.target.value, row });
```
Comment (Pucek9) [inline projects/storefrontlib/src/shared/components/checktable/checktable.component.ts]: sorry, I forgot that for checkbox we have event.target.checked

Comment (architworld) [reply projects/storefrontlib/src/shared/components/checktable/checktable.component.ts]: fixed now

**Comment by Pucek9** [comment]
done by #6686

**Pull request closed without merging**